php/Blockonomics.php

```diff
@@ -79,20 +92,118 @@ public function get_price($currency, $crypto)
+        }
+        elseif (wp_remote_retrieve_response_code($response)==401)
+            $error_str = __($error_crypto.'API Key is incorrect', 'blockonomics-bitcoin-payments');
+        elseif (wp_remote_retrieve_response_code($response)!=200)
+            $error_str = $error_crypto.$response->data;
+        return $error_str;
+    }
+
+    public function check_get_callbacks_response_body ($response, $crypto){
+        $error_str = '';
+        $error_crypto = strtoupper($crypto).' error: ';
+        $response_body = json_decode(wp_remote_retrieve_body($response));
+
+        $callback_secret = get_option('blockonomics_callback_secret');
+        $api_url = WC()->api_request_url('WC_Gateway_Blockonomics');
+        $callback_url = add_query_arg('secret', $callback_secret, $api_url);
+        $callback_url_without_schema = preg_replace('/https?:\/\//', '', $callback_url);
+
+        if (!isset($response_body) || count($response_body) == 0)
+        {
+            $error_str = __($error_crypto.'You have not entered an xPub', 'blockonomics-bitcoin-payments');
+        }
+        elseif (count($response_body) == 1)
+        {
+            $response_callback = '';
+            $response_address = '';
+            if(isset($response_body[0])){
+                $response_callback = isset($response_body[0]->callback) ? $response_body[0]->callback : '';
+                $response_address = isset($response_body[0]->address) ? $response_body[0]->address : '';
+            }
+            $response_callback_without_schema = preg_replace('/https?:\/\//', '', $response_callback);
+            if(!$response_callback || $response_callback == null)
+            {
+                //No callback URL set, set one 
+                $this->update_callback($callback_url, $crypto, $response_address);
+            }
+            elseif($response_callback_without_schema != $callback_url_without_schema)
+            {
+                $base_url = get_bloginfo('wpurl');
+                $base_url = preg_replace('/https?:\/\//', '', $base_url);
+                // Check if only secret differs
+                if(strpos($response_callback, $base_url) !== false)
+                {
+                    //Looks like the user regenrated callback by mistake
+                    //Just force Update_callback on server
+                    $this->update_callback($callback_url, $crypto, $response_address);
+                }
+                else
+                {
+                    $error_str = __($error_crypto."You have an existing callback URL. Refer instructions on integrating multiple websites", 'blockonomics-bitcoin-payments');
+                }
+                
+            }
+        }
+        else 
+        {
+            $error_str = __("You have an existing callback URL. Refer instructions on integrating multiple websites", 'blockonomics-bitcoin-payments');
```

Comment:
$error_crypto should also prefix this error message.

Reply:
Yes, I've moved it to the correct line.

Reply:
The error message does not indicate which crypto the error is for. The error string should be `$error_str = __($error_crypto."You have an existing callback URL. Refer instructions on integrating multiple websites", 'blockonomics-bitcoin-payments');`

Reply:
Sorry, my mistake. Now it's fixed.